storage/json/json.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package json
+
+import (
+	"encoding/json"
+	"flag"
+	"net"
+	"os"
+
+	info "github.com/google/cadvisor/info/v1"
+	storage "github.com/google/cadvisor/storage"
+)
+
+func init() {
+	storage.RegisterStorageDriver("json", new)
+}
+
+type jsonStorage struct {
+	infoField   string
+	machineName string
+	connection  net.Conn
```

Comment:
RE: testing - if you change this type to a `io.WriteCloser` then you can test `AddStats` with a simple mock (e.g. bytes buffer) rather than messing with a test connection.